File: Java/old/1 Two Sum/code1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int left = 0;
        int right = nums.length - 1;
        int[] sorted = Arrays.copyOf(nums, nums.length);
        Arrays.sort(sorted);
        while(right > left) {
            if(sorted[left] + sorted[right] == target) {
                for(int i = 0; i < nums.length; i++) {
                    if(sorted[left] == nums[i]) {
                        left = i;
                        break;
                    }
                }
                for(int i = 0; i < nums.length; i++) {
                    if(sorted[right] == nums[i] && i != left) {
                        right = i;
                        break;
                    }
                }
                return new int[] {left, right};
            }
            if(sorted[left] + sorted[right] > target) right--;
            if(sorted[left] + sorted[right] < target) left++;
        }
        return null;
    }
}

File: C++/old/1 Two Sum/code1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> m;
        for(int i = 0; i < nums.size(); i++)
        {
            if(m.count(target - nums[i]) != 0) return {m[target - nums[i]], i};
            m[nums[i]] = i;
        }

        return {};
    }
};
